tt_metal/impl/device/device.cpp

```diff
@@ -305,12 +306,7 @@ bool Device::initialize(const std::vector<uint32_t>& l1_bank_remap) {
                         return std::make_pair(producer_core, consumer_core);
                     });
 
-            this->sysmem_manager = std::make_unique<SystemMemoryManager>(
-                this->id_,
-                dispatch_cores,
-                [&, this](CoreCoord core) { return this->worker_core_from_logical_core(core); }
-            );
-
+            SystemMemoryManager& manager = SystemMemoryManager::instance(this);
```

Comment:
since you are uplfiting sysmem manager to accept device, I can move dispatch core manager to be owned by device in https://github.com/tenstorrent-metal/tt-metal/pull/4615

Reply:
I still haven't figured out why Sysmem Manager needs to be a global after looking at the existing changes. Are the changes that require this not in yet?

Reply:
@TT-billteng We needed some way for the command queue manager to hold onto the command queue programs/kernels. Unfortunately, those data structures are not defined by the time we get to device. After hearing feedback that we want to make the manager global anyways, I thought the best approach would be to make it global and assign the programs to it in device.cpp.

Reply:
we can eventually make dispatch_core_manager owned by device but it depends on having a device pool

Reply:
Cleaned it up. Owned by device again.